### DISCUSSION TITLE
CPU Utilization Near 100% During Inference - Optimization Suggestions?

### DISCUSSION THREAD

**Discussion by BeanWei** [urchade/GLiNER#153] [Q&A]
First of all, thank you for providing this great NER library and for making it open source. I really appreciate your hard work and dedication.

I am new to this field and have been experimenting with GLiNER for a project. During inference on CPU, I noticed that the CPU utilization is consistently near 100%.

Could you please provide any suggestions or guidance on how to optimize CPU usage during inference? Any tips or recommended configurations would be highly appreciated.

**Comment by ori-laminar** [comment]
@BeanWei Are you running with a relatively low threshold? Try running with the highest threshold you can that still returns the entities you are looking for.
The CPU time might come from processing redundant entities with low scores.
